go-server-server/main.go

```diff
@@ -15,73 +15,146 @@ import (
     "crypto/tls"
     "crypto/x509"
     "io/ioutil"
-    "time"
     "os"
     "os/signal"
     "syscall"
     "context"
+    "sync"
+    "time"
 )
 
+const CERT_MONITOR_FREQUENCY = 3600 * time.Second
+
 func StartHttpServer(handler http.Handler) {
     log.Printf("info: http endpoint started")
     log.Fatal(http.ListenAndServe(":8090", handler))
 }
 
-
-func StartHttpsServer(handler http.Handler) {
-    clientCert, err := ioutil.ReadFile(*sw.ClientCertFlag)
-    if err != nil {
-        log.Fatalf("error: couldn't open client cert file, %s", err)
-    }
-    clientCertPool := x509.NewCertPool()
-    clientCertPool.AppendCertsFromPEM(clientCert)
-
-    // Setup HTTPS client cert the server trust and validation policy
-    tlsConfig := &tls.Config{
-        ClientCAs: clientCertPool,
-        // NoClientCert
-        // RequestClientCert
-        // RequireAnyClientCert
-        // VerifyClientCertIfGiven
-        // RequireAndVerifyClientCert
-        ClientAuth: tls.RequireAndVerifyClientCert,
-        MinVersion: tls.VersionTLS12,
-    }
-
-    tlsConfig.BuildNameToCertificate()
-
-    server := &http.Server{
-        Addr:      ":8081",
-        Handler:   handler,
-        TLSConfig: tlsConfig,
+func StartHttpsServer(handler http.Handler, messenger <-chan int, wgroup *sync.WaitGroup) {
+    defer wgroup.Done()
```

Comment:
With Os.exit, defer won't run, right?, so is this expected to run?

Reply:
No, this is not expected to run. Done() is just an elegant way of having the main function wait for the go-routines. I implemented this to remove the infinite loop waiting in the main function used earlier.

Reply:
Ok, got it.